src/alerts.rs

```diff
@@ -22,12 +23,15 @@ use std::{
 
 pub(crate) type ForkProof<H, D, S> = (UncheckedSignedUnit<H, D, S>, UncheckedSignedUnit<H, D, S>);
 
-#[derive(Debug, Decode, Encode)]
+#[derive(Debug, Decode, Encode, Derivative)]
+#[derivative(PartialEq, Eq, Hash)]
```

Comment:
Why do we derive Hash here (and for several other types)? Or: where is it used?

Reply:
`Output` is being held in `HashMap`s in the added tests.